src/core/atom.ts

```diff
@@ -17,7 +17,7 @@ type Write<Update> = (
   update: Update
 ) => void | Promise<void>
 
-type WithInitialValue<Value> = {
+export type WithInitialValue<Value> = {
```

Comment:
not exporting this is intentional in #488 as being implementation details. Can we test it without exporting it? I think you can just define it in the test code.

Reply:
Or, as it's not a public type, we don't need to test it?

Reply:
Then I can't test the type of `atom()`

Reply:
I think what we want to test is `atom(0)` has `PrimitiveAtom<number>` type. That's it. WithInitialValue is implementation details and should be tested as behavior.